src/xwayland/xwm/surface.rs

```diff
@@ -829,6 +851,24 @@ impl X11Surface {
         }
     }
 
+    fn read_window_property_u32(&self, atom: impl Into<Atom>) -> Result<Option<u32>, ConnectionError> {
+        let conn = self.conn.upgrade().ok_or(ConnectionError::UnknownError)?;
+        let reply = match conn
+            .get_property(false, self.window, atom, AtomEnum::CARDINAL, 0, 1)?
+            .reply_unchecked()
+        {
+            Ok(Some(reply)) => reply,
+            Ok(None) | Err(ConnectionError::ParseError(_)) => return Ok(None),
+            Err(err) => return Err(err),
+        };
+
+        let Some(u32) = reply.value32().unwrap().next() else {
```

Comment:
`u32` is a valid identifier? Certainly interesting, if that is the case, but please use something like `value` instead.